### PR TITLE
Fix for issue 620.  Modified to account for coupon_id in PromoCodes

### PR DESCRIPTION
**Pull request by cshift**

### CODE REVIEW

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php [hunk 1]
```diff
@@ -2282,9 +2282,19 @@ public function addResendFilter($collection, $magentoStoreId, $itemType)
                     $this->logError($this->__('The item type sent in the filter does not match any of the available options.'));
             }
             if ($resendTurn) {
-                $collection->addFieldToFilter('entity_id', array('lteq' => $lastItemSent));
+                if($itemType == Ebizmarts_MailChimp_Model_Config::IS_PROMO_CODE) {
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Please change this to stop repeating the same condition multiple times.
Adding a variable and setting it in the switch could do the same.

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php [hunk 2]
```diff
@@ -2282,9 +2282,19 @@ public function addResendFilter($collection, $magentoStoreId, $itemType)
...
+                    $collection->addFieldToFilter('coupon_id', array('lteq' => $lastItemSent));
+                }
+                else {
+                    $collection->addFieldToFilter('entity_id', array('lteq' => $lastItemSent));
+                }
             } else {
-                $collection->addFieldToFilter('entity_id', array('gt' => $lastItemSent));
+                if($itemType == Ebizmarts_MailChimp_Model_Config::IS_PROMO_CODE) {
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Please change this to stop repeating the same condition multiple times.
Adding a variable and setting it in the switch could do the same.

**Comment by Santiagoebizmarts** [comment]
Thanks for your contribution @cshift 
We appreciate your help.

Regards.

**Comment by centerax** [comment]
closes https://github.com/mailchimp/mc-magento/issues/620

**Pull request merged**